miteD/middleware/api.py

```diff
@@ -42,10 +70,52 @@ def stop(self):
                     with suppress(asyncio.CancelledError):
                         self._loop.run_until_complete(task)
                 self._loop.close()
+                self._deregister_with_consul()
 
             def get_remote_service(self, service_name, version):
                 return RemoteService(name=service_name, version=version, nc=self._nc)
 
+            def _register_with_consul(self):
+                self.consul_id = os.getenv("HOSTNAME")
+                # This is also the pod name
+
+                api_data = {
+                  "Name": self.consul_id,
+                  "ID": self.consul_id,
+                  "Address": f"api/{name}/{versions[0]}",
+                  "Tags": [
+                      "api",
+                      f"{versions[0]}",
```

Comment:
do we really need f-string for these one variable strings?

Reply:
Not at all. Would `str()` be better? Granted most times versions are already strings.